### ISSUE TITLE
getArtistTopTrack Function get 400 Bad Request

### ISSUE THREAD

**Issue by amrfarid140** [kaaes/spotify-web-api-android#82]
When I try to do the following code : 

 protected List<Track> doInBackground(String... params) {
            try {
                artistID = params[0];
                Log.v("Passed Id",artistID);
                Tracks tracksPager   = spotify.getArtistTopTrack(this.artistID);
                List<Track> tracks = new ArrayList<>();
                for (Track track : tracksPager.tracks) {
                    Track localTrack = track;
                    tracks.add(localTrack);
                }

                return tracks;
            }catch (Exception e){
                Log.v("Execute issue",e.toString());
            }
            return null;
        }

gives me : 

retrofit.RetrofitError: 400 Bad Request

I am sure that the artist ID I am passing is A valid one for example : 13ubrt8QOOCPljQ2FL1Kca

**Comment by JohnnyCrazy** [comment]
It seems like the country parameter [is required now](https://developer.spotify.com/web-api/get-artists-top-tracks/) (it wasn't before as far as I remember?)  
Try the following:
````java
protected List doInBackground(String... params) 
{
    try 
    {
        artistID = params[0];
        Log.v("Passed Id",artistID);
        Map<String, Object> options = new Map<String, Object>();
        options.put("country", "YOUR_COUNTRY"); //Replace here
        Tracks tracksPager = spotify.getArtistTopTrack(this.artistID, options);
        List tracks = new ArrayList<>();
        for (Track track : tracksPager.tracks) 
        {
            Track localTrack = track;
            tracks.add(localTrack);
        }
        return tracks;
    }catch (Exception e) 
    {
        Log.v("Execute issue",e.toString());
    }
    return null;
}
````

**Comment by amrfarid140** [comment]
I added the country parameter and still getting the same error ..any ideas ? 

        protected List<Track> doInBackground(String... params) {
            try {
                artistID = params[0];
                Log.v("Passed Id",artistID);

                Map<String, Object> options = new Hashtable<String, Object>();
                options.put("country", "uk");
                Tracks tracksPager   = spotify.getArtistTopTrack(this.artistID,spotifyTopTrackQueryParams);
                List<Track> tracks = new ArrayList<>();
                for (Track track : tracksPager.tracks) {
                    Track localTrack = track;
                    tracks.add(localTrack);
                }

                return tracks;
            }catch (Exception e){
                Log.v("Execute issue",e.toString());
            }
            return null;
        }

**Comment by JohnnyCrazy** [comment]
> Tracks tracksPager   = spotify.getArtistTopTrack(this.artistID,spotifyTopTrackQueryParams);
should be
> Tracks tracksPager   = spotify.getArtistTopTrack(this.artistID, options);

**Comment by amrfarid140** [comment]
Oh yes even this gives the same error also ...

![image](https://cloud.githubusercontent.com/assets/5002609/8336170/580d5bbc-1a9a-11e5-818f-3005e2ee53c0.png)

**Comment by JohnnyCrazy** [comment]
Try changing the `uk` to `UK`.

**Comment by amrfarid140** [comment]
unfortunately Same with uk, UK and United Kingdom

**Comment by JohnnyCrazy** [comment]
Then I'm out of options :sweat:

**Comment by amrfarid140** [comment]
Thank you for your help anyway  :D

**Comment by JohnnyCrazy** [comment]
But, you could further analyze the error, what is the output of this code snippet:
````java
protected List doInBackground(String... params) 
{
    try 
    {
        artistID = params[0];
        Log.v("Passed Id",artistID);
        Map<String, Object> options = new HashTable<String, Object>();
        options.put("country", "YOUR_COUNTRY"); //Replace here
        Tracks tracksPager = spotify.getArtistTopTrack(this.artistID, options);
        List tracks = new ArrayList<>();
        for (Track track : tracksPager.tracks) 
        {
            Track localTrack = track;
            tracks.add(localTrack);
        }
        return tracks;
    }catch (RetrofitError error) 
    {
        ErrorDetails details = SpotifyError.fromRetrofitError(error).getErrorDetails();
        System.out.println(details.status + ":" + details.message);
    }
    return null;
}
````

**Comment by amrfarid140** [comment]
06-24 06:53:25.102    2831-2912/? I/deatils﹕ status 400, message Invalid country code

hmmm..any ideas one what should be the right one ?

**Comment by JohnnyCrazy** [comment]
Oh, just noticed UK isn't a valid code, it's just reserverd, try it with GB
![img](http://i.imgur.com/0irr8OL.png)

**Comment by amrfarid140** [comment]
yeah but you know what Instead of UK I used US and it is working 
![image](https://cloud.githubusercontent.com/assets/5002609/8337487/5fd5cf88-1aa3-11e5-9fa9-14adf91b98eb.png)

notice here that UK isn't an option in countries

**Comment by amrfarid140** [comment]
Thanks for you effort and help

**Comment by kaaes** [comment]
Good catch with the country parameter! We'll fix it soon. Feel free to open a PR if you'd like :)

@JohnnyCrazy you're correct about the country code: https://api.spotify.com/v1/artists/0TnOYISbd1XYRBk9myaseg/top-tracks?country=UK vs https://api.spotify.com/v1/artists/0TnOYISbd1XYRBk9myaseg/top-tracks?country=GB

For future reference, we have this nifty console that makes testing different requests quite simple: https://developer.spotify.com/web-api/console/get-artist-top-tracks/

**Comment by JMPerez** [comment]
This should be fixed with #83
